File: tests/smoke/recv.php
#!/usr/bin/env php
<?php

include("proton.php");

$messenger = new Messenger();
$messenger->incoming_window = 10;
$message = new Message();

$address = $argv[1];
if (!$address) {
  $address = "~0.0.0.0";
}
$messenger->subscribe($address);

$messenger->start();

while (true) {
  $messenger->recv();
  $messenger->get($message);
  print "Got: $message\n";
  $messenger->accept();
}

$messenger->stop();

?>

File: tests/smoke/recv.py
#!/usr/bin/env python
from __future__ import print_function
import sys
from proton import *

messenger = Messenger()
messenger.incoming_window = 1
message = Message()

address = "~0.0.0.0"
if len(sys.argv) > 1:
  address = sys.argv[1]
messenger.subscribe(address)

messenger.start()

while True:
  messenger.recv()
  messenger.get(message)
  print("Got: %s" % message)
  messenger.accept()

messenger.stop()
